source/common/buffer/buffer_impl.cc

```diff
@@ -588,5 +588,35 @@ std::vector<Slice::SliceRepresentation> OwnedImpl::describeSlicesForTest() const
   return slices;
 }
 
+size_t OwnedImpl::addFragments(absl::Span<const absl::string_view> fragments) {
+  size_t total_size_to_write = 0;
+
+  for (const auto& fragment : fragments) {
+    total_size_to_write += fragment.size();
+  }
+
+  if (slices_.empty()) {
+    slices_.emplace_back(Slice(total_size_to_write, account_));
+  }
+
+  Slice& back = slices_.back();
```

Comment:
Add TODO to improve performance as discussed in the comments, rather than the downgrade strategy we have here? You can add here the detail about contiguous memory also, that you provided in that comment thread.

Reply:
get it.